cluster/engine.go

```diff
@@ -217,8 +217,13 @@ func (e *Engine) StartMonitorEvents() {
 		if err := <-ec; err != nil {
 			log.WithFields(log.Fields{"name": e.Name, "id": e.ID}).WithError(err).Error("error monitoring events, will restart")
 			if !strings.Contains(err.Error(), "EOF") {
-				// failing node reconnect should use back-off strategy
-				<-e.refreshDelayer.Wait(e.getFailureCount())
+				// failing node reconnect should use back-off strategy to avoid frequent reconnect
+				retryInterval := e.getFailureCount() + 1
```

Comment:
Oh, I see. So this change gets rid of the random backoff logic?

https://github.com/docker/swarm/blob/master/cluster/engine.go#L99

I don't really have any idea how useful that random backoff logic is, but I feel like it might be nice to keep around.